src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs

```diff
@@ -337,5 +337,42 @@ public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)
                 throw new XunitException($"Expected: {string.Join(", ", expected)}{Environment.NewLine}Actual: {string.Join(", ", actual)}");
             }
         }
+
+        public delegate void AssertThrowsActionReadOnly<T>(ReadOnlySpan<T> span);
+
+        public delegate void AssertThrowsAction<T>(Span<T> span);
+
+        // Cannot use standard Assert.Throws() when testing Span - Span and closures don't get along.
+        public static void AssertThrows<E, T>(ReadOnlySpan<T> span, AssertThrowsActionReadOnly<T> action) where E : Exception
+        {
+            try
+            {
+                action(span);
+                throw new XunitException($"Expected exception: : {typeof(E).GetType()} -> Actual: No exception thrown");
+            }
+            catch (E)
+            {
+            }
+            catch (Exception wrongException)
+            {
+                throw new XunitException($"Wrong exception thrown: Expected exception: : {typeof(E).GetType()} -> Actual: {wrongException.GetType()}");
+            }
+        }
```

Comment:
This is actually AssertThrowsAny (it accepts derived types).  To match Assert.Throws' behavior it should look like

https://github.com/dotnet/corefx/blob/9b2f7a834691a3bbad2949181eec707dd4b5a13b/src/System.Text.Json/tests/JsonTestHelper.cs#L706-L732

Both with ThrowsException (which builds the correct message) and with checking an exact match on ex.GetType().

Reply:
@bartonjs, should this be `ThrowsException`, and not `XunitException` that @safern mentioned above (https://github.com/dotnet/corefx/pull/39186#discussion_r300794061)?